### PR TITLE
Vendor upgrade

### PR DESCRIPTION
**Pull request by vsiddharth**
* Re-vendoring dependencies according to the new style. 
* Fixed test failures
* Updated scripts + Makefile

### CODE REVIEW

File: ecs-init/Godeps/Godeps.json [hunk 1]
```diff
@@ -1,22 +1,240 @@
...
+			"Rev": "f561c5e57575bb1e0a2167028b7339b3a8d16fb4"
+		},
+		{
+			"ImportPath": "github.com/cihub/seelog/archive/zip",
+			"Comment": "v2.6-31-gf561c5e",
+			"Rev": "f561c5e57575bb1e0a2167028b7339b3a8d16fb4"
+		},
+		{
+			"ImportPath": "github.com/docker/docker/api/types",
+			"Comment": "docs-v1.12.0-rc4-2016-07-15-6275-g80013fd59",
```
Comment (samuelkarp) [inline ecs-init/Godeps/Godeps.json]: This looks like a really weird branch.  Can you check what you have in your GOPATH and see if we can get something like 17.03.1-ce here?

File: ecs-init/VERSION [hunk 2]
```diff
@@ -1 +0,0 @@
-1.14.1
```
Comment (samuelkarp) [inline ecs-init/VERSION]: Why is this removed?

Comment (vsiddharth) [inline ecs-init/VERSION]: It was accidental. I'll restore it.

File: ecs-init/Godeps/Godeps.json (hunk 1 above)
Comment (vsiddharth) [inline ecs-init/Godeps/Godeps.json]: This is something `godep` performed. 
Do we really need to manually check and update all the vendor dependencies ?

Comment (cabbruzzese) [inline ecs-init/Godeps/Godeps.json]: This file is being included by godeps as a dependency of go-dockerclient. I don't think we should try to change it.

@vsiddharth did you have an existing copy of github.com/docker/docker/ in your $GOPATH/src before using godep save? Wouldn't hurt to go get go-dockerclient again to make sure.

Comment (vsiddharth) [inline ecs-init/Godeps/Godeps.json]: @cabbruzzese I did not have an existing copy of `go-dockerclient` while updating these dependencies. I had a a fresh `go-workspace` when I tried this.

Comment (petderek) [inline ecs-init/Godeps/Godeps.json]: @samuelkarp  agent uses a similarly-named branch of docker, probably for the same reason.

**Comment by vsiddharth** [comment]
@cabbruzzese @petderek Could you have another look at this PR ?
I've resolved the merge conflicts and made minor changes to a few build scripts. 

Thanks

File: ecs-init/Godeps/Godeps.json (hunk 1 above)
Comment (samuelkarp) [inline ecs-init/Godeps/Godeps.json]: @vsiddharth @cabbruzzese @petderek We should figure out exactly why this is happening and fix it both here and in the agent.
